src/napari_matplotlib/scatter.py

```diff
@@ -222,3 +222,124 @@ def _on_update_layers(self) -> None:
         # reset the axis keys
         self._x_axis_key = None
         self._y_axis_key = None
+
+
+class FeaturesHistogramWidget(NapariMPLWidget):
+    n_layers_input = Interval(1, 1)
+    # All layers that have a .features attributes
+    input_layer_types = (
+        napari.layers.Labels,
+        napari.layers.Points,
+        napari.layers.Shapes,
+        napari.layers.Tracks,
+        napari.layers.Vectors,
+    )
+
+    def __init__(self, napari_viewer: napari.viewer.Viewer):
+        super().__init__(napari_viewer)
+        self.axes = self.canvas.figure.subplots()
+
+        self._key_selection_widget = magicgui(
+            self._set_axis_keys,
+            x_axis_key={"choices": self._get_valid_axis_keys},
+            call_button="plot",
+        )
+        self.layout().addWidget(self._key_selection_widget.native)
+
+        self.update_layers(None)
+
+    def clear(self) -> None:
+        """
+        Clear the axes.
+        """
+        self.axes.clear()
+
+        self.layout().addWidget(self._key_selection_widget.native)
+
+    @property
+    def x_axis_key(self) -> Optional[str]:
+        """Key to access x axis data from the FeaturesTable"""
+        return self._x_axis_key
+
+    @x_axis_key.setter
+    def x_axis_key(self, key: Optional[str]) -> None:
+        self._x_axis_key = key
+        self._draw()
+
+    def _set_axis_keys(self, x_axis_key: str) -> None:
+        """Set both axis keys and then redraw the plot"""
+        self._x_axis_key = x_axis_key
+        self._draw()
+
+    def _get_valid_axis_keys(
+        self, combo_widget: Optional[ComboBox] = None
```

Comment:
I don't think this combo_widget argument is needed

Reply:
Fixed in dcab365f898cb945207395af2f0e84afd9170dff

Reply:
Fixed in dcab365f898cb945207395af2f0e84afd9170dff